airflow/ui/src/components/ui/ActionButton.tsx

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { type ButtonProps, IconButton } from "@chakra-ui/react";
+import type { FC, ReactElement } from "react";
+
+import { Button, Tooltip } from "src/components/ui";
+
+type Props = {
+  readonly actionName: string;
+  readonly colorPalette?: string;
+  readonly icon: ReactElement;
+  readonly onClick: () => void;
+  readonly text: string;
+  readonly variant?: string;
+  readonly withText?: boolean;
+} & ButtonProps;
+
+const ActionButton = ({
+  actionName,
+  colorPalette,
+  icon,
+  onClick,
+  text,
+  variant = "outline",
+  withText = true,
+}: Props) => {
+  const ButtonComponent: FC<ButtonProps> = withText ? Button : IconButton;
+
+  return (
+    <Tooltip content={actionName} disabled={Boolean(withText)}>
+      <ButtonComponent
+        aria-label={actionName}
+        colorPalette={withText ? colorPalette : "blue"}
+        onClick={onClick}
+        size={withText ? "md" : "sm"}
+        variant={withText ? variant : "ghost"}
```

Comment:
`variant` and `colorPalette` (or any other custom styling for 'withText') I don't like but couldn't find an elegant way.

Meaning that it's not super easy to use an `<ActionButton />` and customize style for `withText` and `compact` (withText=false) version distinctly.

Reply:
The final `TriggerDAGButton` and `ClearRunButton` props are clean and easy to use. Making sure those actions have consistant display over different places, it's just that intermediate layer that I don't really like.

Reply:
I am not a pro in React... is some kind of inheritance also possible (compared to use an abstract component in another component as middle layer)?

Reply:
That's not really critical. Maybe someone will find a more flexible way. We don't really need it for now.

Moving forward.

Reply:
If things didin't change there's no inheritance in react but Composition. I tried thinking about that but I didn't find an ideal approach.